### PR TITLE
[EOS-19957][EOS-19964] adds provisioner resources implementation

### PR DESCRIPTION
**Pull request by andkononykhin2**
That PR solves two things:

- provides new resource based implementation for provisioner resources installation on a set of nodes
- makes automation for sw upgrade testing infra preparation much easier

New resources are described in updated [resource.md](https://github.com/andkononykhin2/cortx-prvsnr/blob/EOS-19957_EOS-19964_provisioner-resources/docs/Development/resources.md) doc.

-----
[View rendered docs/Development/resources.md](https://github.com/andkononykhin2/cortx-prvsnr/blob/EOS-19957_EOS-19964_provisioner-resources/docs/Development/resources.md)

### CODE REVIEW

File: api/python/provisioner/commands/validator/validator.py [hunk 1]
```diff
@@ -348,22 +347,11 @@ def validate(self, path: Path):
...
-
-        with open(path, 'rb') as fh:
-            while True:
-                data = fh.read(4096)
-                if not data:
-                    break
-                hash_method.update(data)
-
-        if not compare_digest(hash_method.digest(), self.hash_sum):
+        if not compare_digest(hash_obj.digest(), self.hash_sum):
```
Comment (ypise) [inline api/python/provisioner/commands/validator/validator.py:352]: Please add a comment here mentioning what this digest is and how we are using it in this context of code.

File: api/python/provisioner/config.py [hunk 2]
```diff
@@ -583,6 +585,7 @@ class HashType(Enum):
 
     MD5 = "md5"
     SHA256 = "sha256"
+    SHA384 = "sha384"
```
Comment (ypise) [inline api/python/provisioner/config.py:588]: Why this odd one?
Ideally we should stick with the more secure **_sha512_** only.

**Review by ypise — APPROVED** [review]
Overall no issues.
Just a few suggestions.

File: api/python/provisioner/config.py (hunk 2 above)
Comment (andkononykhin2) [reply api/python/provisioner/config.py:588]: Ok. I set 512 as a default. But I am not sure that ts would be more secure in all the cases. Please check [that](https://en.wikipedia.org/wiki/Hash_function_security_summary), length extension attacks part..

File: api/python/provisioner/commands/validator/validator.py (hunk 1 above)
Comment (andkononykhin2) [reply api/python/provisioner/commands/validator/validator.py:355]: done

**Pull request merged**